src/platform/graphics/webgl/webgl-texture.js

```diff
@@ -299,6 +300,16 @@ class WebglTexture {
 
         const requiredMipLevels = texture.requiredMipLevels;
 
+        if (texture._array) {
+            // for texture arrays we reserve the space in advance
+            gl.texStorage3D(gl.TEXTURE_2D_ARRAY,
+                            1,
```

Comment:
we should handle mipmaps as well: 
- here when creating the texture
- when we upload and it's specified
- if it's not specified we should generate mipmaps

Reply:
I have handled the mipmaps generation or uploading.
I have updated the example to provide a toggle for showing mipmaps.